1_nwisdata/src/nwis_pull.R

```diff
@@ -0,0 +1,308 @@
+        paste(
+          "gd_put(",
+          "remote_ind=target_name,",
+          "local_source='%s',",
+          "mock_get=I('move'),",
+          "on_exists=I('update'))",
+          sep="\n      "),
+        scipiper::as_ind_file(file.path(folders$tmp, sprintf('%s_siteinfo.feather', task_name))))
+    }
+  )
+
+  retrieve_data <- scipiper::create_task_step(
+    step_name = 'retrieve_data',
+    target_name = function(task_name, step_name, ...) {
+      file.path(folders$out, sprintf('%s.feather', task_name))
+    },
+    command = function(task_name, target_name, ...) {
+      sprintf(
+        paste(
+          "gd_get(",
+          "ind_file='%s')",
+          sep="\n      "),
+        scipiper::as_ind_file(target_name))
+    }
+  )
+
+  retrieve_siteinfo <- scipiper::create_task_step(
+    step_name = 'retrieve_siteinfo',
+    target_name = function(task_name, step_name, ...) {
+      file.path(folders$out, sprintf('%s_siteinfo.feather', task_name))
+    },
+    command = function(task_name, target_name, ...) {
+      sprintf(
+        paste(
+          "gd_get(",
+          "ind_file='%s')",
+          sep="\n      "),
+        scipiper::as_ind_file(target_name))
+    }
+  )
+
+  task_plan <- scipiper::create_task_plan(
+    task_names=sort(partitions$PullTask),
+    task_steps=list(partition, download, confirm_download, extract_data, extract_siteinfo,
+                    post_data, post_siteinfo, retrieve_data, retrieve_siteinfo),
+    final_steps=c('post_data', 'post_siteinfo'),
+    add_complete=TRUE,
+    ind_dir=folders$log)
+
+}
+
+create_nwis_pull_makefile <- function(makefile, task_plan) {
+  create_task_makefile(
+    makefile=makefile, task_plan=task_plan,
+    include='1_nwisdata.yml',
+    packages=c('dplyr', 'dataRetrieval', 'feather', 'scipiper'),
+    file_extensions=c('ind','feather'))
+}
+
+get_nwis_data <- function(ind_file, partition, wq_dates) {
+
+  # prepare the argument to pass to readNWISdata
+  nwis_args <- list(
+    characteristicName=partition$Params[[1]]$CharacteristicNames,
+    startDateLo=wq_dates$startDate,
+    startDateHi=wq_dates$endDate
+  )
+  # specify sites either by whole state and siteType (if we didn't split the
+  # site into multiple site sets) or as a specific list of sites (otherwise)
+  if(partition$NumPulls == 1) {
+    nwis_args <- c(nwis_args, list(
+      statecode=partition$StateCode,
+      siteType=partition$SiteType[[1]]$SiteType
+    ))
+  } else {
+    nwis_args <- c(nwis_args, list(
+      siteid=partition$Sites[[1]]$Site
+    ))
+  }
+
+  # do the data pull
+  nwis_dat_time <- system.time({
+    nwis_dat <- do.call(dataRetrieval::readNWISdata, nwis_args)
+  })
+  message(
+    "NWIS pull for ", partition$PullTask,
+    " took ", nwis_dat_time['elapsed'], " seconds and",
+    " returned ", nrow(nwis_dat), " rows")
+
+  # make nwis_dat a tibble, converting either from data.frame (the usual case) or
+  # NULL (if there are no results)
+  nwis_dat <- as_data_frame(nwis_dat)
```

Comment:
the comment here says `make nwis_dat a tibble`, but the code suggests that you are converting it into a data.frame.

Reply:
`data_frame` == `tibble`...or more precisely, `data_frame` == `tbl_df` and the `tibble` package is so named because `tbl` sounds like "tibble". So that's all I meant; was specifying data_frame rather than data.frame.
```
> x <- data_frame(a=1)
> class(x)
[1] "tbl_df"     "tbl"        "data.frame"
```

Reply:
ah OK, because I know `as_tibble` is a function, so I wasn't sure if you meant to use that instead

Reply:
gotchya. I did just confirm from the tibble docs that `data_frame()` is an exact alias to `tibble()`